src/psyclone/configuration.py

```diff
@@ -802,10 +809,58 @@ def __init__(self, config, section):
                 # Handled in the base class APISpecificConfig
                 pass
+            elif key == "grid-properties":
+                # Field properties have the format:
+                # go_grid_area_u: {0}%%grid%%area_u: array,
+                # First the name, then the Fortran code to access the property
+                # then the type ("array" or "scalar")
+                all_props = self.create_dict_from_string(section[key])
+                for grid_property in all_props:
+                    try:
+                        access, field_type = all_props[grid_property]\
+                                            .split(":")
+                    except ValueError:
+                        # Raised when the string does not contain exactly
+                        # two values separated by ":"
+                        error = "Invalid property \"{0}\" found with value " \
+                                "\"{1}\" in \"{2}\". It must have exactly " \
+                                "two ':'' separated values." \
+                                .format(grid_property,
+                                        all_props[grid_property],
+                                        config.filename)
+                        raise ConfigurationError(error)
+                    # Make sure to remove the spaces which the config
+                    # file might contain
+                    self._grid_properties[grid_property] = \
+                        (access.strip(), field_type.strip())
+                # Check that the required values for xstop and ystop
+                # are defined:
+                for required in ["go_grid_xstop", "go_grid_ystop",
+                                 "go_grid_data",
+                                 "go_grid_internal_inner_start",
+                                 "go_grid_internal_inner_stop",
+                                 "go_grid_internal_outer_start",
+                                 "go_grid_internal_outer_stop",
+                                 "go_grid_whole_inner_start",
+                                 "go_grid_whole_inner_stop",
+                                 "go_grid_whole_outer_start",
+                                 "go_grid_whole_outer_stop"]:
+                    if required not in self._grid_properties:
+                        error = "The config file {0} does not contain " \
+                                "values for \"{1}\".".format(config.filename,
+                                                             required)
+                        raise ConfigurationError(error)
             else:
                 raise ConfigurationError("Invalid key \"{0}\" found in "
                                          "\"{1}\".".format(key,
                                                            config.filename))
+    # ---------------------------------------------------------------------
+    @property
+    def field_properties(self):
```

Comment:
I think this property should be named `grid_properties`?

Reply:
I'll discuss this at tonight's Telco.

Reply:
See #676: for now I renamed everything to be grid properties, and we can decide later if and how to split this into grid- and field-properties. This way full backwards compatibility for this patch is maintained.

Reply:
Great.